### ISSUE TITLE
Publish documentation for old scalafmt versions to the website

### ISSUE THREAD

**Issue by charego** [scalameta/scalafmt#2575]
Currently the website documents the latest and greatest [configuration options](https://scalameta.org/scalafmt/docs/configuration.html).
It is not possible to review older versions of scalafmt.

Workarounds:
- checkout the project and [run the site locally](https://scalameta.org/scalafmt/docs/contributing-website.html#running-the-site-locally) with your desired version
- checkout the project and browse the source code for your version

From a usability perspective, it would be nice if the website published some older versions of the documentation.
It's not always possible/feasible to use the bleeding edge version.

Visual example:

<img width="168" alt="http4s" src="https://user-images.githubusercontent.com/5209296/120712189-55ded880-c48e-11eb-96b1-6e27313116ac.png">

**Comment by alejandrohdezma** [comment]
I remember @olafurpg mentioned [this feature](https://twitter.com/olafurpg/status/1223889336451637249) from the Docusaurus plugin last year. If that's still the case, it could be a matter of removing 👇🏼 from the `build.sbt`.

https://github.com/scalameta/scalafmt/blob/5d02fda3982182a9962394b93f20fef76631eeec/build.sbt#L255

Although that would only solve the problem for future versions.

**Comment by olafurpg** [comment]
The Scalafmt website already documents which settings are only available in recent versions with the "Since VERSION" header above section. @charego can you elaborate on what exact information you struggled to find from the current website? I think it's best if the live website has sufficient coverage to support older versions because then you avoid situations like

- top results on Google point to outdated versions (I found this quite frustrating when working with Playframework and Akka)
- users send PRs fixing wording of sentences in outdated versions, when the latest version has the same sentence

@alejandrohdezma it's right that updating that setting will configure the build to publish the website to Maven Central on every release. It could be worth enabling. Other Scalameta projects like MUnit have used this feature from the start and you can browse old versions on https://javadoc.io/doc/org.scalameta/munit-docs_2.12/0.7.21/munit/docs/reports.html

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/1408093/134817465-edc09472-0f8a-41e0-8488-fba886f66164.png">

Most users probably don't know this even exists but that could be fixed by making it more discoverable. Also, the search bar doesn't work because it uses a remote index (algolia) which only indexes the main website but it's maybe not a huge problem.

**Comment by charego** [comment]
@olafurpg I found it difficult to interpret (and make changes to) an existing `.scalafmt.conf` file that was based on an older version of Scalafmt. Some of the options had been removed from Scalafmt and did not appear on the website, so in order to learn about those options I needed to use the workarounds listed above.

Another website that does this well is [Mill's documentation](https://com-lihaoyi.github.io/mill/). Even if search results bring you to an outdated version, you'd be able to fix that by selecting your version from the dropdown.

<img width="1090" alt="mill-docs" src="https://user-images.githubusercontent.com/5209296/135551951-633d7610-ee64-4837-9d2c-95585fc5f82a.png">
